d1/arch/sdl/joy.c

```diff
@@ -199,6 +255,8 @@ void joy_init()
 			}
 			for (j=0; j < SDL_Joysticks[num_joysticks].n_hats; j++)
 			{
+				if (Joystick.n_buttons >= MAX_BUTTONS_PER_JOYSTICK)
```

Comment:
It looks like we need to check that there's space in `joybutton_text` for 4 more entries, not just one (2 more for axis buttons).

Reply:
Good catch, I've fixed the check.